hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/physical/CreateDagVisitor.java

```diff
@@ -126,7 +145,12 @@ public Vertex onProject(ProjectPhysicalRel rel) {
                 (BiFunctionEx<Function<Object[], Object[]>, Object[], Object[]>) Function::apply
         ));
 
-        connectInput(rel.getInput(), vertex, null);
+        Consumer<Edge> edgeConfig = null;
+        if (rel.getTraitSet().getCollation().getFieldCollations().size() > 0) {
+            edgeConfig = edge -> edge.distributeTo(localMemberAddress).allToOne("");
+        }
```

Comment:
Maybe we could do something more generic here. Just 2 quick ideas:
- push project into sort
- preserving parallelism from upstream operators

Reply:
Our Project/Filter implementation destroys the collation, that's the issue. I agree we should rather use `isolated` edge and an equal parallelism with the upstream, that would be more generic. A distributed edge is more expensive, even though in this particular case it doesn't matter because the upstream was also distributed to the same member.

For Filter we should check that there's no collation - but not with an assert because we're never sure which possible plans did we miss.